### PR TITLE
Port the port_drayage_plugin to ROS2

### PR DESCRIPTION
**Pull request by JonSmet**
# PR Details
## Description

This PR includes updates to port the port_drayage_plugin node to ROS2 and to launch it from the guidance.launch.py launch file.

Additionally, this PR includes other small fixes:
- Classes within the port_drayage_plugin package had a mix of snake_case and camelCase function names. This PR has updated all functions to camelCase to match the style used throughout CARMA Platform.
- Unused code from early port_drayage_plugin development was removed from the package. Initially, this package was planned to be a strategic plugin within the CARMA Platform guidance subsystem, but was later adjusted in order to primarily process received port drayage-related Mobility Operation messages (i.e. generate a route to a port drayage destination received from infrastructure) and broadcast port drayage-related Mobility Operation messages (i.e. arrival messages).

## Related Issue

Issue #1839 

## How Has This Been Tested?

All unit tests pass; carma-platform was built and port_drayage_plugin node was integration tested in a development environment with received Mobility Operation messages.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: port_drayage_plugin/src/port_drayage_plugin.cpp [hunk 1]
```diff
@@ -13,290 +13,131 @@
...
-            ROS_DEBUG_STREAM("Set Active Route service call was not successful.");
-            return false;
-        }
+    else {
+      // No response was received after making the service call
+      RCLCPP_DEBUG_STREAM(get_logger(), "Set Active Route service call was not successful.");
+      return false;
     }
 
-    bool PortDrayagePlugin::plan_maneuver_cb(cav_srvs::PlanManeuversRequest &req, cav_srvs::PlanManeuversResponse &resp){
```
Comment (kjrush) [inline port_drayage_plugin/src/port_drayage_plugin.cpp:145]: This (the maneuver planning functions here) was just dead code previously, correct?

File: port_drayage_plugin/include/port_drayage_plugin/port_drayage_state_machine.hpp [hunk 2]
```diff
@@ -25,10 +26,11 @@ namespace port_drayage_plugin
      * 
      * INACTIVE = PortDrayagePlugin is not currently operating in an active
      *  capacity
-     * EN_ROUTE = The vehicle is currently EN_ROUTE to a destination, either
-     *  under the direciton of the PortDrayagePlugin or otherwise.
-     * AWAITING_DIRECTION = The vehicle is currently stopped under the command
-     *  of the PortDrayagePlugin and is awaiting further guidance from the 
+     * EN_ROUTE_TO_INITIAL_DESTINATION = The vehicle is currently enroute  to its initial
```
[the remark this answers is not in the document]
Comment (JonSmet) [reply port_drayage_plugin/include/port_drayage_plugin/port_drayage_state_machine.hpp:29]: Yes, the corresponding PortDrayageState enum is updated accordingly.

File: port_drayage_plugin/include/port_drayage_plugin/port_drayage_plugin_config.hpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  /**
+   * \brief Struct containing the algorithm configuration values for port_drayage_plugin
+   */
+  struct Config
+  {
+    std::string cmv_id = ""; // The CMV's static ID, which is its license plate
+
+    std::string cargo_id = ""; // The ID of the cargo that is being carried at system start-up. Empty if CMV is not carrying cargo.
+
+    bool enable_port_drayage = false; // Flag to activate/deactivate port drayage operations. If false, the port drayage state machine will remain INACTIVE
```
[the remark this answers is not in the document]
Comment (msmcconnell) [reply port_drayage_plugin/include/port_drayage_plugin/port_drayage_plugin_config.hpp:34]: I think this kind of thing is harder to standardize an API for, so we could add plugin discovery but beyond that having a different type might not be overly meaningful. If we wanted to take that small step though, that could be ok.

File: port_drayage_plugin/include/port_drayage_plugin/port_drayage_worker.hpp [hunk 4]
```diff
@@ -82,152 +86,129 @@ namespace port_drayage_plugin
...
-            std::function<void(cav_msgs::MobilityOperation)> _publish_mobility_operation;
-            std::function<void(cav_msgs::UIInstructions)> _publish_ui_instructions;
-            std::function<bool(cav_srvs::SetActiveRoute)> _set_active_route;
-            std::shared_ptr<lanelet::projection::LocalFrameProjector> _map_projector = nullptr;
-            bool _starting_at_staging_area; // Flag indicating CMV's first destination; 'true' indicates Staging Area Entrance; 'false' indicates Port Entrance.
-            bool _enable_port_drayage; // Flag to enable to port drayage operations. If false, state machine will remain in 'INACTIVE' state
+            std::shared_ptr<carma_planning_msgs::msg::RouteEvent> latest_route_event_ = nullptr;
+            PortDrayageStateMachine pdsm_;
+            std::string previously_completed_operation_;
+            const OperationID pickup_operation_{OperationID::PICKUP};
```
[the remark this answers is not in the document]
Comment (JonSmet) [reply port_drayage_plugin/include/port_drayage_plugin/port_drayage_worker.hpp]: Good question, I don't believe there was a good reason as to why these instantiated member variables were created during initial implementation. I think the code reads better using the enum directly, as you suggested, so I have updated the code and re-verified unit test results.

File: port_drayage_plugin/launch/port_drayage_plugin_launch.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+from launch.actions import DeclareLaunchArgument
+from launch.substitutions import LaunchConfiguration
+from carma_ros2_utils.launch.get_current_namespace import GetCurrentNamespace
+
+import os
+
+
+'''
+This file is can be used to launch the CARMA port_drayage_plugin_node.
+  Though in carma-platform it may be launched directly from the base launch file.
```
[the remark this answers is not in the document]
Comment (JonSmet) [reply port_drayage_plugin/launch/port_drayage_plugin_launch.py]: Fixed.

**Review by kjrush — APPROVED** [review]

**Pull request merged**